File: app/src/main/java/cn/leo/magicscreenadapter/MainActivity.java
package cn.leo.magicscreenadapter;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

/**
 * @author Leo
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        addFragment();
        addFragmentV4();
    }

    private void addFragment() {
        FragmentManager fm = getSupportFragmentManager();
        fm.beginTransaction()
                .replace(R.id.container, new TestFragment())
                .commitAllowingStateLoss();
    }

    private void addFragmentV4() {
        androidx.fragment.app.FragmentManager fm = getSupportFragmentManager();
        fm.beginTransaction()
                .replace(R.id.container1, new TestV4Fragment())
                .commitAllowingStateLoss();
    }
}

File: app/src/main/java/cn/leo/magicscreenadapter/MainActivity.kt
package cn.leo.magicscreenadapter

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author Leo
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        addFragment()
        initRecyclerView()
    }

    private fun addFragment() {
        val fm = supportFragmentManager
        fm.beginTransaction()
                .replace(R.id.container, TestV4Fragment())
                .commitAllowingStateLoss()
    }

    private fun initRecyclerView() {
        rv1.layoutManager = LinearLayoutManager(this)
        val rvAdapter = RvAdapter()
        rv1.adapter = rvAdapter
        rvAdapter.data = arrayListOf(1, 2, 3, 4, 5, 6, 7, 8, 9)
    }
}
